sdk/core/Azure.Core/src/Shared/RetriableStream.cs

```diff
@@ -108,6 +108,13 @@ public override async Task<int> ReadAsync(byte[] buffer, int offset, int count,
 
             private async Task RetryAsync(Exception exception, bool async, CancellationToken cancellationToken)
             {
+                // Depending on the timing, the stream can be closed as a result of cancellation when the transport closes the stream.
+                // If the user requested cancellation, we translate to TaskCanceledException, similar to what we do HttpWebRequestTransport.
+                if (exception is ObjectDisposedException)
```

Comment:
Is there any reason this exception could be thrown other than the reason we are targeting and should that still propagate as is?

Reply:
We are only translating if the user requested cancellation, so if this exception occurs when the user didn't cancel, it would still get propagated as is.